lib/Driver/ToolChains.cpp

```diff
@@ -318,6 +314,33 @@ ToolChain::constructInvocation(const CompileJobAction &job,
   addCommonFrontendArgs(*this, context.OI, context.Output, context.Args,
                         Arguments);
 
+  // Pass along an -import-objc-header arg, optionally replacing the header name
+  // with a temporary PCH name, if there was an input to this job that
+  // generated a temporary PCH from the header in question.
```

Comment:
I think this is overly conservative. It should be fine to just pass along *any* PCH outputs, and if there aren't any then we forward as is.